packages/amplication-client/src/routes/routesUtil.tsx

```diff
@@ -1,8 +1,29 @@
 import React, { Suspense } from "react";
-import { Redirect, Route, Switch } from "react-router-dom";
+import { match, Redirect, Route, Switch } from "react-router-dom";
 import { RouteDef } from "./appRoutes";
 import useAuthenticated from "../authentication/use-authenticated";
 
+export enum MatchRoute {
+  WORKSPACE = "workspace",
+  PROJECT = "project",
+  RESOURCE = "resource",
+  COMMIT_ID = "commitId",
+  ENTITY_ID = "entityId",
+  FIELD_ID = "fieldId",
+  BUILD_ID = "buildId",
+}
+
+export type AppMatchRoute = {
+  match: match<{ [property in MatchRoute]: string }>;
+};
```

Comment:
by doind this you assume you have all the parameters in all the pages and that is not correct. 
especially for the use which use it, they will get a type with all the properties even when some of them are empty.
I don't think this change make sense. also- I don't see any reason why you would include this change in this PR